src/Our.Umbraco.MeganavV8.Core/ValueConnectors/MeganavV8ValueConnector.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Deploy;
+using Umbraco.Core.Models;
+
+namespace Our.Umbraco.MeganavV8.Core.ValueConnectors
+{
+    /// <summary>
+    /// Represents a value connector for the Our.Umbraco.MeganavV8 property editor.
+    /// </summary>
+    /// <seealso cref="IValueConnector" />
+    public class MeganavV8ValueConnector : IValueConnector
+    {
+
+        public string ToArtifact(object value, PropertyType propertyType, ICollection<ArtifactDependency> dependencies)
+        {
+            var svalue = value as string;
+            if (string.IsNullOrWhiteSpace(svalue) || !svalue.DetectIsJson())
+            {
+                return null;
+            }
+
+            var rootLinks = ParseLinks(JArray.Parse(svalue), dependencies, Direction.ToArtifact);
+
+            return rootLinks.ToString(Formatting.None);
+        }
+
+        public object FromArtifact(string value, PropertyType propertyType, object currentValue)
+        {
+            var rootLinks = ParseLinks(JArray.Parse(value), null, Direction.FromArtifact);
+
+            value = rootLinks.ToString(Formatting.None);
```

Comment:
There's not need to parse links when restoring the value.

Reply:
Good point!

After sleeping on it (it was a late one last night), I agree with you 👍 

This has been removed